filament/backend/test/test_Blit.cpp

```diff
@@ -0,0 +1,260 @@
+    for (auto rt : dstRenderTargets)  api.destroyRenderTarget(rt);
+}
+
+TEST_F(BackendTest, ColorMinify) {
+    auto& api = getDriverApi();
+
+    constexpr int kSrcTexWidth = 1024;
+    constexpr int kSrcTexHeight = 1024;
+    constexpr auto kSrcTexFormat = filament::backend::TextureFormat::RGBA8;
+
+    // Create a SwapChain and make it current. We don't really use it so the res doesn't matter.
+    auto swapChain = api.createSwapChainHeadless(256, 256, 0);
+    api.makeCurrent(swapChain, swapChain);
+
+    // Create a source texture.
+    Handle<HwTexture> srcTexture = api.createTexture(
+        SamplerType::SAMPLER_2D, kNumLevels, kSrcTexFormat, 1, kSrcTexWidth, kSrcTexHeight, 1,
+        TextureUsage::SAMPLEABLE | TextureUsage::UPLOADABLE | TextureUsage::COLOR_ATTACHMENT);
+    createBitmap(api, srcTexture, kSrcTexWidth, kSrcTexHeight, 0, float3(0.5, 0, 0));
+    createBitmap(api, srcTexture, kSrcTexWidth, kSrcTexHeight, 1, float3(0, 0, 0.5));
+
+    // Create a destination texture.
+    Handle<HwTexture> dstTexture = api.createTexture(
+        SamplerType::SAMPLER_2D, kNumLevels, kDstTexFormat, 1, kDstTexWidth, kDstTexHeight, 1,
+        TextureUsage::SAMPLEABLE | TextureUsage::COLOR_ATTACHMENT);
+
+    // Create a RenderTarget for each texture's miplevel.
+    Handle<HwRenderTarget> srcRenderTargets[kNumLevels];
+    Handle<HwRenderTarget> dstRenderTargets[kNumLevels];
+    for (uint8_t level = 0; level < kNumLevels; level++) {
+        srcRenderTargets[level] = api.createRenderTarget( TargetBufferFlags::COLOR,
+                kSrcTexWidth >> level, kSrcTexHeight >> level, 1, { srcTexture, level, 0 }, {}, {});
+        dstRenderTargets[level] = api.createRenderTarget( TargetBufferFlags::COLOR,
+                kDstTexWidth >> level, kDstTexHeight >> level, 1, { dstTexture, level, 0 }, {}, {});
+    }
+
+    // Do a "magnify" blit from level 1 of the source RT to the level 0 of the desination RT.
+    const int srcLevel = 1;
+    api.blit(TargetBufferFlags::COLOR0, dstRenderTargets[0],
+            {0, 0, kDstTexWidth, kDstTexHeight}, srcRenderTargets[srcLevel],
+            {0, 0, kSrcTexWidth >> srcLevel, kSrcTexHeight >> srcLevel}, SamplerMagFilter::LINEAR);
+
+    // Push through an empty frame to allow the texture to upload and the blit to exectue.
+    getDriverApi().beginFrame(0, 0);
+    getDriverApi().commit(swapChain);
+    getDriverApi().endFrame(0);
+
+    // Grab a screenshot.
+    getDriverApi().beginFrame(0, 0);
+    dumpScreenshot(api, dstRenderTargets[0], "ColorMinify.png");
+    getDriverApi().commit(swapChain);
+    getDriverApi().endFrame(0);
+
+    // Wait for the ReadPixels result to come back.
+    api.finish();
+    executeCommands();
+    getDriver().purge();
+
+    // Check if the image matches perfectly to our golden run.
+    const uint32_t expected = 0xe2353ca6;
+    printf("Computed hash is 0x%8.8x, Expected 0x%8.8x\n", sPixelHashResult, expected);
```

Comment:
This seems quite brittle to rely on a global static field, esp. if we add more tests. Couldn't you instead pass a reference of some kind to dumpScreenshot to it writes the result in a safe place?

Reply:
Yes this was sloppy.  I added a commit to the PR to fix this.